### PR TITLE
Promote {aarch64,i686,x86_64}-unknown-uefi to Tier 2

### PR DESCRIPTION
**Pull request by nicholasbishop**
MCP: https://github.com/rust-lang/compiler-team/issues/555

CC @dvdhrm

### CODE REVIEW

**Review by JohnTitor — APPROVED** [review]

**Comment by JohnTitor** [comment]
I'm going to r+ as the MCP has been completed and this follows the same way as https://github.com/rust-lang/rust/pull/95705.
just in case cc @oli-obk as you seconded the MCP

@bors r+

File: src/doc/rustc/src/platform-support/unknown-uefi.md [hunk 1]
```diff
@@ -72,28 +72,14 @@ target = ["x86_64-unknown-uefi"]
...
-
-Alternatively, you can install `cargo-xbuild` via
-`cargo install --force cargo-xbuild` and build for the UEFI targets via:
-
-```sh
-cargo \
-    +nightly \
-    xbuild \
-    --target x86_64-unknown-uefi
+```text
```
Comment (dvdhrm) [inline src/doc/rustc/src/platform-support/unknown-uefi.md:78]: Was the change from `sh` to `text` intentional? The comments should work fine in `sh`, shouldn't it?

**Review by dvdhrm — APPROVED** [review]
Thanks! Looks good!

File: src/doc/rustc/src/platform-support/unknown-uefi.md (hunk 1 above)
Comment (nicholasbishop) [reply src/doc/rustc/src/platform-support/unknown-uefi.md]: Good catch, fixed. (I had copy-pasted that from https://github.com/rust-lang/rust/pull/95705/files#diff-7a14c39a7e2dc831476df3687623b15a8ac53450b9fe34769b1fba1fe4d16ef9R54)

**Comment by GuillaumeGomez** [comment]
I think this is the one which failed in https://github.com/rust-lang/rust/pull/104143#issuecomment-1307254273. Please don't hesitate to re-r+ if my guess was wrong.

@bors r-

**Comment by nicholasbishop** [comment]
It looks like the job that failed was `dist-aarch64-msvc`, with bad arguments passed to `armasm`. This PR shouldn't have affected `aarch64-msvc`. Maybe the `cc` upgrade PR was the problem?

@rustbot ready

**Comment by GuillaumeGomez** [comment]
I guess we'll know soon enough. I made another rollup with the `cc` update there: https://github.com/rust-lang/rust/pull/104152

**Comment by nicholasbishop** [comment]
The new rollup without this PR failed, looks like it's the same `aarch64-msvc` error as before. So I think this PR can be re-approved.

**Comment by GuillaumeGomez** [comment]
Absolutely!

@bors r=JohnTitor

**Pull request merged**